src/filtering.rs

```diff
@@ -118,9 +161,114 @@ impl SnapshotFilter {
             }
         }
 
+        if matches!(&self.filter_after, Some(after) if snapshot.time <= after.0)
+            || matches!(&self.filter_before, Some(before) if snapshot.time >= before.0)
+            || matches!((&self.filter_size,&snapshot.summary), (Some(size),Some(summary)) if !size.matches(summary.total_bytes_processed))
+            || matches!((&self.filter_size_added,&snapshot.summary), (Some(size),Some(summary)) if !size.matches(summary.data_added))
+        {
+            return false;
+        }
```

Comment:
Can we add a comment here what the implied logic should be, so I can review against it?

Reply:
added